packages/cli/src/utils.ts

```diff
@@ -32,25 +32,33 @@ export function baseName(fileName: string) {
  * @returns file mtime in ms, or null if the file does not exist or
  * is not a file.
  */
+async function statFile(filePath: PathLike): Promise<Stats | null> {
+  try {
+    return await fsp.stat(filePath);
+  } catch (e) {
+    return null;
+  }
+}
+
 export async function mtime(filePath: PathLike): Promise<number | null> {
   // NB: 'ctime' is actually the time that the stats of the file were last changed.
   // And 'birthtime' is not guaranteed across platforms.
   // Therefore mtime is the most reliable indicator of when the file was created,
   // especially since we write files atomically (e.g. by moving them into place after writing them
   // as temp files).
 
-  let fileStat: Stats;
-  try {
-    fileStat = await fsp.stat(filePath);
-  } catch (e) {
-    return null;
-  }
-  if (!fileStat.isFile()) {
+  const fileStat = await statFile(filePath);
+  if (!fileStat || !fileStat.isFile()) {
     return null;
   }
   return fileStat.mtimeMs;
 }
 
+export async function isFile(filePath: PathLike): Promise<boolean | null> {
```

Comment:
Nitpick, but do we need to return `null` here? It won't fit neatly into a boolean without the caller casting the result.

Reply:
No, we don't. I'll fix it.